File: src/sensor_temperatura.cpp
#include "sensor_temperatura.hpp"

#include <utility>

SensorTemperatura::SensorTemperatura(std::string tag, double valorInicial)
    : tag_(std::move(tag)), valor_(valorInicial) {
}

bool SensorTemperatura::atualizar(double novoValor) {
    if (novoValor < -50.0 || novoValor > 80.0) return false;
    valor_ = novoValor;
    return true;
}

double SensorTemperatura::valor() const { return valor_; }
const std::string& SensorTemperatura::tag() const { return tag_; }

File: src/sensor_temperatura.py
class SensorTemperatura:
    def __init__(self, tag: str, valor: float) -> None:
        self._tag = tag
        self._valor = valor

    def atualizar(self, novo_valor: float) -> bool:
        if not -50.0 <= novo_valor <= 80.0:
            return False
        self._valor = novo_valor
        return True

    @property
    def valor(self) -> float:
        return self._valor

    @property
    def tag(self) -> str:
        return self._tag
